src/rules/order-imports.ts

```diff
@@ -178,21 +182,24 @@ function isPlainRequireModule(node): boolean {
 	);
 }
 
-function isPlainImportModule(node: NodeOrToken): boolean {
-	return node.type === 'ImportDeclaration' && node.specifiers != null && node.specifiers.length > 0;
+function isAllowedImportModule(node: NodeOrToken, context): boolean {
+	const unassignedImportsAllowed = getOptions(context).unassignedImports === 'allow';
+	const hasNodeSpecifier = node.specifiers != null && node.specifiers.length > 0;
+
+	return node.type === 'ImportDeclaration' && (hasNodeSpecifier || unassignedImportsAllowed);
 }
 
-function canCrossNodeWhileReorder(node: NodeOrToken): boolean {
-	return isPlainRequireModule(node) || isPlainImportModule(node);
+function canCrossNodeWhileReorder(node: NodeOrToken, context): boolean {
+	return isPlainRequireModule(node) || isAllowedImportModule(node, context);
```

Comment:
I suggest leaving the `isPlainImportModule` function alone and instead, on this line, add in a check for 

I _think_ it would be: 
```javascript
return getOptions(context).unassignedImports === 'allow'` || isPlainRequireModule(node) || isAllowedImportModule(node)
```

Reply:
not 100% sure on this, I think this would also make it sort on variables or other types of nodes

Reply:
@Tibfib Yeah, I was thinking that too but wanted to make sure I don't let any other node which may not be an import pass through.